nixos/modules/system/boot/stage-1.nix

```diff
@@ -688,7 +688,7 @@ in
 
   config = mkIf config.boot.initrd.enable {
     assertions = [
-      { assertion = any (fs: fs.mountPoint == "/") fileSystems;
+      { assertion = !config.boot.initrd.systemd.enable -> any (fs: fs.mountPoint == "/") fileSystems;
```

Comment:
If `config.boot.initrd.systemd.enable` is set, there is no assertion. If `config.boot.initrd.systemd.enable` is not set, there is an assertion when no "/" mountpoint is set.